website/server/controllers/api-v3/tasks.js

```diff
@@ -653,6 +656,49 @@ api.scoreTask = {
       user.save(),
       task.save(),
     ];
+
+    // Handle shared completion
+    if (task.type === 'todo' && direction === 'up' && task.group && task.group.sharedCompletion === 'singleCompletion') {
+      let masterTask = await Tasks.Task.findOne({
+        _id: task.group.taskId,
+      }).exec();
+
+      masterTask.completed = true;
+      promises.push(masterTask.save());
+
+      const tasksToRemove = await Tasks.Task.find({
+        'group.taskId': task.group.taskId,
+        $and: [
+          {userId: {$exists: true}},
+          {userId: {$ne: user._id}},
+        ],
+      }).exec();
+
+      tasksToRemove.forEach(async (taskToRemove) => {
+        promises.push(taskToRemove.remove());
+      });
+    } else if (task.type === 'todo' && direction === 'up' && task.group && task.group.sharedCompletion === 'allAssignedCompletion') {
+      let allAssignedComplete = await _.reduce(task.group.assignedUsers, async (result, id) => {
```

Comment:
_.reduce doesn't return a promise so I don't think this piece of code is working correctly